### PR TITLE
feat: add job deploy command

### PR DESCRIPTION
**Pull request by arinda-arif**
New command will be introduced:
```
- optimus job deploy [--jobs jobA,jobB] [--namespace sample_namespace]
```

Approach:
```
- Use GetJobSpecifications, AddJobSpecifications, UpdateJobSpecifications API
- Add/Update will be based on the Get response
```

### CODE REVIEW

File: client/cmd/job/deploy.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	cmd.MarkFlagRequired("namespace")
+	return cmd
+}
+
+func (e *deployCommand) PreRunE(_ *cobra.Command, _ []string) error {
+	if err := e.loadConfig(); err != nil {
+		return err
+	}
+	e.logger = logger.NewClientLogger()
+	e.namespaceSurvey = survey.NewNamespaceSurvey(e.logger)
```
Comment (deryrahman) [inline client/cmd/job/deploy.go:70]: if it's not used, we can remove this one

Comment (arinda-arif) [reply client/cmd/job/deploy.go]: removed this

File: client/cmd/job/deploy.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+func (e *deployCommand) getJobSpecByNames(namespaceJobPath string) ([]*model.JobSpec, error) {
+	jobSpecs := make([]*model.JobSpec, 0)
+	for _, jobName := range e.jobNames {
+		jobSpecReadWriter, err := specio.NewJobSpecReadWriter(afero.NewOsFs(), specio.WithJobSpecParentReading())
+		if err != nil {
+			return nil, err
+		}
+		jobSpec, err := jobSpecReadWriter.ReadByName(namespaceJobPath, jobName)
```
Comment (irainia) [inline client/cmd/job/deploy.go:199]: void calling this method like this. instead, read all specs first. after that, filter based on the names to make it more efficient.

**Review by irainia — CHANGES_REQUESTED** [review]
I don't have further context about the changes, but for now the feedback is only this. Subject to an updated context from my side.

File: client/cmd/job/deploy.go (hunk 2 above)
Comment (arinda-arif) [reply client/cmd/job/deploy.go]: updated

**Review by irainia — APPROVED** [review]

**Pull request merged**